core/engine/src/builtins/builder.rs

```diff
@@ -387,12 +387,6 @@ impl BuiltInConstructorWithPrototype<'_> {
         }
 
         let mut object = self.object.borrow_mut();
-        let function = object
-            .downcast_mut::<NativeFunctionObject>()
-            .expect("Builtin must be a function object");
-        function.f = NativeFunction::from_fn_ptr(self.function);
-        function.constructor = Some(ConstructorKind::Base);
-        function.realm = Some(self.realm.clone());
```

Comment:
Oops, I suggested to remove this, but I think this is still needed.

Reply:
I've made some updates to this, take a look